Doctrine/Orm/DataProvider.php

```diff
@@ -61,7 +61,18 @@ public function getItem(ResourceInterface $resource, $id, $fetchData = false)
         $manager = $this->managerRegistry->getManagerForClass($entityClass);
 
         if ($fetchData || !method_exists($manager, 'getReference')) {
-            return $manager->find($entityClass, $id);
+            $repository = $manager->getRepository($entityClass);
+            $queryBuilder = $repository->createQueryBuilder('o');
+            $identifier = $manager->getClassMetadata($resource->getEntityClass())->getIdentifierFieldNames()[0];
+            $queryBuilder->where($queryBuilder->expr()->eq('o.'.$identifier, ':id'))->setParameter('id', $id);
+
+            foreach ($this->extensions as $extension) {
+                if ($extension instanceof QueryItemExtensionInterface) {
+                    $extension->applyToItem($resource, $id, $queryBuilder);
```

Comment:
Btw, can you change signature of extensions both collection and item methods to `ResourceInterface $resource, QueryBuilder $queryBuilder, Request $request` to be coherent with the signature of filters?